### DISCUSSION TITLE
Use as framework - initial migration with setting update.

### DISCUSSION THREAD

**Discussion by hanming2033** [pocketbase/pocketbase#2423] [Q&A]
Hi pocketbase,

Thank you making such a nice product. I am trying to have migration script that configures pocketbase on initial setup.
In the docs, https://pocketbase.io/docs/migrations/, it says that we can get the Settings by using the code below. <br>

        // and even set default app settings<br>
        settings, _ := dao.FindSettings()<br>
        settings.Meta.AppName = "test"<br>
        settings.Smtp.Enabled = false<br>
        if err := dao.SaveSettings(settings); err != nil {<br>
            return err<br>
        }<br>

The problem is that the settings(default) is only persisted to DB after the initial migration scripts are run. So, the above code will fail if used as initial migration script.<br>

To get around it, I used the New func to instantiate the default settings. The good thing is that, once default settings are set in migration script, pocketbase will not try to set it again.<br>

        "github.com/pocketbase/pocketbase/models/settings"<br>
        settings := settings.New()     <= instead of settings, _ := dao.FindSettings()

My 2 cents:
1. initialize default settings before any migration is run OR
2. if my approach is correct, explanation of this in the doc will help.

**Comment by ganigeorgiev** [comment]
Thanks. I wasn't able to reproduce it initially, but I forgot that I wasn't working with blank `pb_data`.

I've added in `master` a small change in the system init migration to persists the default settings `_params` so that they are available when accessed inside the users defined migrations (_it will be available with the next v0.15.3 release sometime later this week_).

**Reply by hanming2033** [reply]
Best support ever! Fixed it in one day! Let me check it out 😄
